superset/db_engine_specs/doris.py

```diff
@@ -0,0 +1,285 @@
+    max_column_name_length = 64
+    default_driver = "pydoris"
+    sqlalchemy_uri_placeholder = (
+        "doris://user:password@host:port/catalog.db[?key=value&key=value...]"
+    )
+    encryption_parameters = {"ssl": "0"}
+    supports_dynamic_schema = True
+
+    column_type_mappings = (  # type: ignore
+        (
+            re.compile(r"^tinyint", re.IGNORECASE),
+            TINYINT(),
+            GenericDataType.NUMERIC,
+        ),
+        (
+            re.compile(r"^largeint", re.IGNORECASE),
+            LARGEINT(),
+            GenericDataType.NUMERIC,
+        ),
+        (
+            re.compile(r"^decimal.*", re.IGNORECASE),
+            types.DECIMAL(),
+            GenericDataType.NUMERIC,
+        ),
+        (
+            re.compile(r"^double", re.IGNORECASE),
+            DOUBLE(),
+            GenericDataType.NUMERIC,
+        ),
+        (
+            re.compile(r"^varchar(\((\d+)\))*$", re.IGNORECASE),
+            types.VARCHAR(),
+            GenericDataType.STRING,
+        ),
+        (
+            re.compile(r"^char(\((\d+)\))*$", re.IGNORECASE),
+            types.CHAR(),
+            GenericDataType.STRING,
+        ),
+        (
+            re.compile(r"^json.*", re.IGNORECASE),
+            types.JSON(),
+            GenericDataType.STRING,
+        ),
+        (
+            re.compile(r"^binary.*", re.IGNORECASE),
+            types.BINARY(),
+            GenericDataType.STRING,
+        ),
+        (
+            re.compile(r"^quantile_state", re.IGNORECASE),
+            QUANTILE_STATE(),
+            GenericDataType.STRING,
+        ),
+        (
+            re.compile(r"^agg_state.*", re.IGNORECASE),
+            AGG_STATE(),
+            GenericDataType.STRING,
+        ),
+        (
+            re.compile(r"^hll", re.IGNORECASE),
+            HLL(),
+            GenericDataType.STRING
+        ),
+        (
+            re.compile(r"^bitmap", re.IGNORECASE),
+            BITMAP(),
+            GenericDataType.STRING,
+        ),
+        (
+            re.compile(r"^array.*", re.IGNORECASE),
+            ARRAY(),
+            GenericDataType.STRING,
+        ),
+        (
+            re.compile(r"^map.*", re.IGNORECASE),
+            MAP(),
+            GenericDataType.STRING,
+        ),
+        (
+            re.compile(r"^struct.*", re.IGNORECASE),
+            STRUCT(),
+            GenericDataType.STRING,
+        ),
+        (
+            re.compile(r"^datetime.*", re.IGNORECASE),
+            types.DATETIME(),
+            GenericDataType.STRING,
+        ),
+        (
+            re.compile(r"^date.*", re.IGNORECASE),
+            types.DATE(),
+            GenericDataType.STRING,
+        ),
+        (
+            re.compile(r"^text.*", re.IGNORECASE),
+            TEXT(),
+            GenericDataType.STRING,
+        ),
+        (
+            re.compile(r"^string.*", re.IGNORECASE),
+            String(),
+            GenericDataType.STRING,
+        ),
+
```

Comment:
Would you mind removing these blank lines? Actually if you run `tox -e pre-commit` locally then `black` should clean this up.

Reply:
OK I will modify it, thanks for your suggestion